pkg/apiserver/registry/core/seed/strategy.go

```diff
@@ -197,3 +202,38 @@ func syncLegacyAccessRestrictionLabelWithNewFieldOnUpdate(seed, oldSeed *core.Se
 		seed.Spec.AccessRestrictions = removeAccessRestriction(seed.Spec.AccessRestrictions, "eu-access-only")
 	}
 }
+
+func syncBackupSecretRefAndCredentialsRef(backup *core.SeedBackup) {
+	if backup == nil {
+		return
+	}
+
+	emptySecretRef := corev1.SecretReference{}
+
+	// secretRef is set and credentialsRef is not, sync both fields.
+	if backup.SecretRef != emptySecretRef && backup.CredentialsRef == nil {
```

Comment:
I guess it would be better/cleaner to make `SecretRef` a pointer type now, no?

Reply:
I started this way but it caused some troubles with the managedseed API related to the seedTemplate being encoded into it, thus reverted to the value and used empty secretRef.

Reply:
OK, fair enough. If we plan to remove this field in 3 versions anyways, we can live with it